gen/installer/bash.py

```diff
@@ -638,12 +638,15 @@ def fill_template(base_name, format_args):
 def make_dcos_launch():
     # NOTE: this needs to be kept in sync with build_dcos_launch.sh
     work_dir = py.path.local.mkdtemp()
+    git_src = 'file://{}'.format(os.getcwd())
+    if os.path.exists('ext/upstream'):
+        git_src = os.path.join(git_src, 'ext/upstream')
+    # clone so the in-place build doesn't affect this tree
+    subprocess.check_call(['git', 'clone', '--progress', git_src, str(work_dir)])
```

Comment:
No local changes can be cloned in git, so we should actually be validating there aren't any. That's what the GitLocal source fetcher does: https://github.com/dcos/dcos/blob/master/pkgpanda/build/src_fetchers.py#L129

Otherwise _magically_ in the user's build any local / uncommitted changes they have won't show up, which is painful.

In terms of code, writing two functions which do the same thing is duplication. Each one can have bugs. Writing a couple lines that does the "simple clone" is duplicating the clone code. Git's api, and the `--mirror` actually is critical for correctness in some situations, there is a reason it was added to `fetch_git` (And can actually dig through the commit history if you want to find out why).